include/hipSYCL/sycl/libkernel/accessor.hpp

```diff
@@ -1097,19 +1098,21 @@ class accessor
     auto data = data_mobile_ptr.get_shared_ptr();
     assert(data);
 
-    if(sizeof(dataT) != data->get_element_size())
-      assert(false && "Accessors with different element size than original "
-                      "buffer are not yet supported");
-
     rt::dag_node_ptr node;
     {
       rt::dag_build_guard build{rt::application::dag()};
 
-      auto explicit_requirement =
-          rt::make_operation<rt::buffer_memory_requirement>(
+      std::unique_ptr<rt::buffer_memory_requirement> explicit_requirement;
+      if(sizeof(dataT) != data->get_num_elements())
```

Comment:
```suggestion
      if(sizeof(dataT) != data->get_element_size())
```
I don't understand this - shouldn't it be as in the suggestion?